docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/microsoft-graph-security-api-source.md

```diff
@@ -75,14 +75,17 @@ The following steps show you how to create a service application:
 
 1. Request the appropriate [permissions for the application](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-configure-app-access-web-apis#application-permission-to-microsoft-graph). Click on **API Permissions**, then **Add a permission** and select **Microsoft Graph**.
 
-    You need to find and select the **SecurityEvents.Read.All** permission. See [this list](https://docs.microsoft.com/en-us/graph/permissions-reference#security-permissions) of all the available security permissions.
+    You need to find and select the **SecurityAlert.Read.All**, **SecurityAlert.ReadWrite.All**, **SecurityIncident.Read.All**, **SecurityIncident.ReadWrite.All** and **SecurityEvents.Read.All** permissions. See [this list](https://docs.microsoft.com/en-us/graph/permissions-reference#security-permissions) of all the available security permissions.
 
     :::note
     An Administrator must approve (grant) these permissions before the Source will function.
     :::
 
     ![6_ms_graph_app_add_permissions.png](/img/send-data/6_ms_graph_app_add_permissions.png)
 
+1. Also, an Application permission role is required. Click on **App roles**, then **Create app role**. Under this provide a display name for the role. Select **Applications** as the allowed member type. Provide **SecurityEvents.Read.All** as the value and provide a proper description for the same. Select the checkbox to enable the role and click **Apply**.
```

Comment:
1. Follow the steps below to enable **Application** permission role. 
	1. In the **Manage** menu select **App roles**.
	1. Click the **Create app role** tab.
		1. **Display name**. Enter the display name for the role. 
		1. **Allowed member types**. Select **Applications** as the allowed member type. 
		1. **Value**. Enter **SecurityEvents.Read.All** as the value.
		1. **Description**. Enter a brief description. 
		1. Select the checkbox to enable the app role .
	1. Click **Apply**.